src/samples/Teams/TeamsAgent/TeamsAgent.cs

```diff
@@ -0,0 +1,172 @@
+﻿using AdaptiveCards.Rendering;
+using Microsoft.Agents.Builder;
+using Microsoft.Agents.Builder.App;
+using Microsoft.Agents.Builder.App.AdaptiveCards;
+using Microsoft.Agents.Builder.State;
+using Microsoft.Agents.Core.Models;
+using Microsoft.Agents.Extensions.Teams.App;
+using Microsoft.Agents.Extensions.Teams.Connector;
+using Microsoft.Agents.Extensions.Teams.Models;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace TeamsAgent
+{
+    public class TeamsAgent : AgentApplication
+    {
+        IHttpClientFactory _httpClientFactory;
+        public TeamsAgent(AgentApplicationOptions options, IHttpClientFactory httpClientFactory) : base(options)
+        {
+            _httpClientFactory = httpClientFactory;
+            RegisterExtension(new TeamsAgentExtension(this), tae =>
+            {
+                tae.OnMessageEdit(MessageEdited);
+                tae.MessageExtensions.OnQuery("findNuGetPackage", OnQuery);
+                tae.MessageExtensions.OnSelectItem(OnSelectItem);
+            });
+            OnMessage("/help", (t, _, ct) => t.SendActivityAsync("TeamsAgent demo", cancellationToken: ct));
+            OnMessageReactionsAdded(OnMessageReaction);
+            OnConversationUpdate(ConversationUpdateEvents.MembersAdded, WelcomeMessageAsync);
+            OnActivity(ActivityTypes.Message, OnMessageAsync);
+        }
+
+        private async Task<MessagingExtensionResult> OnSelectItem(ITurnContext turnContext, ITurnState turnState, object item, CancellationToken cancellationToken)
```

Comment:
this handler seems to have breaking changes from BF-SDK: https://learn.microsoft.com/en-us/dotnet/api/microsoft.bot.builder.teams.teamsactivityhandler.onteamsmessagingextensionselectitemasync?view=botbuilder-dotnet-stable

1, The previous API requires a MessageExtensionResponse instead of MessageExtensionResult
2. The returned item was a JObject and now it's an object

Reply:
`JObject` is gone.  That is a NewtonSoft type.  This will be an issue elsewhere in Teams and we'll need to document when we develop a migration story.

Not sure about the result/response.  Would need to dig and/or talk to Teams.